### Game data exploration

ARRM database files were copied from ARRM install folder and converted from mdb to csv using mdb-tools

Sources:
- ARRM database files
- SteamDB data
- what else?

Preferred features for a database:
- name/title
- filename (important for arcade titles)
- description
- platform
- release date/year
- genre(s)
- developer
- publisher
- number of players
- cooperative
- rating

In [34]:
import pandas as pd

## DAT Databases

In [35]:
dat_daphne = pd.read_csv('csv/dat_database_daphne.csv')
dat_dos = pd.read_csv('csv/dat_database_dos.csv')
dat_mame = pd.read_csv('csv/dat_database_mame.csv')
dat_scumm_vm = pd.read_csv('csv/dat_database_scummvm.csv')

/tmp/ipykernel_50833/1416037915.py:3: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  dat_mame = pd.read_csv('csv/dat_database_mame.csv')


### DOS

In [36]:
print(dat_dos.shape[0])
dat_dos.head()

7482


,ID,gametitle_mame,developer_mame,year_mame,genre_mame,description_mame,publisher_mame
0,61191,Stunt Island,The Assembly Line,19920101T000000,"Flight Simulator,Vehicle Simulation",Stunt Island was marketed as The Stunt Flying ...,"Walt Disney Computer Software, Inc."
1,61192,Tommy's Yahtzee,Tommy's Toys,19860101T000000,Board / Party Game,Tommy's Yahtzee is a shareware implementation ...,Freeware
2,61193,Fantasy Empires,Silicon Knights,19931001T000000,Strategy,Build and control an Empire! In Fantasy Empire...,"Strategic Simulations, Inc."
3,61194,Pune,Hardware Not Included,19950101T000000,Action,Pune is a light cycle/snake-type game for mult...,Hardware Not Included
4,61195,Mortal Kombat,Probe Software Ltd.,19940525T000000,"Action,Fighting","Five Hundred years ago, an ancient and well re...","Acclaim Entertainment, Inc."


In [37]:
dat_dos.isnull().sum()

ID                   0
gametitle_mame       0
developer_mame       3
year_mame            0
genre_mame           1
description_mame    12
publisher_mame      31
dtype: int64

DOS analysis: A very nice set and i love DOS games. Lets use it.

### Daphne

In [38]:
print(dat_daphne.shape[0])
dat_daphne.head()

208


,ID,gametitle_mame,filename_mame,cloneof_mame,manufacturer_mame,year_mame,genre_mame
0,599,.38 Ambush Alley,38ambushalley,NaN,NaN,-,NaN
1,600,Space Ace,ace,NaN,NaN,-,NaN
2,601,Space Ace,ace91,NaN,NaN,-,NaN
3,602,Space Ace,ace91_euro,NaN,NaN,-,NaN
4,603,Space Ace,aceeuro,NaN,NaN,-,NaN


In [39]:
dat_daphne.isnull().sum()

ID                     0
gametitle_mame         0
filename_mame          0
cloneof_mame         208
manufacturer_mame    208
year_mame              0
genre_mame           208
dtype: int64

Analysis: Daphne dataframe is only 34 rows, nulls for manufacturer and genre info. Lets skip this database.

### MAME

In [40]:
print(dat_mame.shape[0])
dat_mame.head()

252693


,ID,systemname_mame,gametitle_mame,filename_mame,cloneof_mame,manufacturer_mame,year_mame,genre_mame
0,355065,advmame 0.94-RetroPie-260.dat,"PuckMan (Japan set 1, Probably Bootleg)",puckman,NaN,Namco,1980/01/01,NaN
1,355066,advmame 0.94-RetroPie-260.dat,PuckMan (Japan set 2),puckmana,puckman,Namco,1980/01/01,NaN
2,355067,advmame 0.94-RetroPie-260.dat,PuckMan (Japan set 1 with speedup hack),puckmanf,puckman,Namco,1980/01/01,NaN
3,355068,advmame 0.94-RetroPie-260.dat,Puckman (Falcom),puckmanh,puckman,hack,1980/01/01,NaN
4,355069,advmame 0.94-RetroPie-260.dat,Pac-Man (Midway),pacman,puckman,[Namco] (Midway license),1980/01/01,NaN


In [41]:
dat_mame['systemname_mame'].value_counts()

systemname_mame
MAME 0.260.dat                                   46199
MAME 0.240 (Arcade).dat                          36604
mame2016 - MAME 0.174 Arcade XML.dat             32071
mame2015 - MAME 0.160 XML.dat                    30906
Mame 0.239.xml                                   29027
Mame 0.270.xml                                   17538
Mame 0.267.xml                                   12904
mame2010.xml                                      8833
FinalBurn_Neo_ClrMame_Pro_XML_Arcade_only.dat     6998
advmame 12-106.dat                                6166
advmame 0.94-RetroPie-260.dat                     5563
mame2003-plus.xml                                 5219
mame2003.xml                                      4726
vgmplay.xml                                       3645
FB Alpha 2012 v0.2.97.30.dat                      3369
mame2000 mame4all - MAME 0.37b5.dat               2241
FB Alpha v0.2.96.71 working_roms.dat               684
Name: count, dtype: int64

In [42]:
dat_mame.isnull().sum()

ID                        0
systemname_mame           0
gametitle_mame            0
filename_mame             0
cloneof_mame         101633
manufacturer_mame     11907
year_mame             95935
genre_mame           193224
dtype: int64

Analysis: The MAME list is huge (222251 rows), genre missing for 87%, year for 42% of the titles. No descriptions. However the existence of mame filenames could be useful. <br>
Columns to keep: gametitle_mame, filename_mame manufacturer_mame and year_mame.

### ScummVM

In [43]:
print(dat_scumm_vm.shape[0])
dat_scumm_vm.head()

483


,ID,gametitle_mame,filename_mame,cloneof_mame,manufacturer_mame,year_mame,genre_mame
0,44251,The 11th Hour The Sequel to The 7th Guest,11h,NaN,Trilobyte,-,NaN
1,44252,Five Lethal Demons,5ld,NaN,Off Studio,-,NaN
2,44253,Five Magical Amulets,5ma,NaN,Off Studio,-,NaN
3,44254,Amazon - Guardians Of Eden,access,NaN,NaN,-,NaN
4,44255,Putt-Putt And Fatty Bear's Activity Pack,activity,NaN,Humongous Entertainment,-,NaN


In [44]:
dat_scumm_vm.isnull().sum()

ID                     0
gametitle_mame         0
filename_mame          0
cloneof_mame         483
manufacturer_mame    149
year_mame              0
genre_mame           318
dtype: int64

SCUMM analysis: Bit of a niche dataset but still useful. Manufacturer and genre missing for 35% titles but after inspection the 35% represents more niche games or some non-SCUMM games. Let's come back to this later.

## GameTDB

In [45]:
gamestdb = pd.read_csv('csv/games_on_gametdb.csv')
print(gamestdb.shape[0])
gamestdb.head()

53858


,N°,gametdb_id,gametdb_type,gametdb_region,gametdb_languages,gametdb_title_de,gametdb_title_en,gametdb_title_es,gametdb_title_fr,gametdb_title_it,...,gametdb_players,gametdb_genre,gametdb_rom,gametdb_rom_sans_ext,gametdb_platform,gametdb_title_cn,gametdb_title_tw,gametdb_synopsis_cn,gametdb_synopsis_tw,gametdb_rom_cleaned
0,1108862,A22J,3DS,NTSC-J,JP,Bokujou Monogatari Futago no Mura+,Bokujou Monogatari Futago no Mura+,NaN,Bokujou Monogatari Futago no Mura+,Bokujou Monogatari Futago no Mura+,...,1.0,NaN,Bokujou Monogatari Futago no Mura+ (Japan) (JA...,Bokujou Monogatari Futago no Mura+ (Japan) (JA),3DS,Bokujou Monogatari Futago no Mura+,Bokujou Monogatari Futago no Mura+,NaN,NaN,Bokujou Monogatari Futago no Mura+
1,1108863,A2AA,3DS,NaN,"EN,FR,DE,ES,IT,JP,KR,TW",Pokémon Ultrasonne: Update,Pokémon Ultra Sun,Pokémon Ultra Sun Actualización Ver. 1.2,Pokémon Ultra-Soleil: Mise à jour,Pokémon Ultrasole: Aggiornamenti,...,1.0,"role-playing,action rpg","Pokémon Ultra Sun (EN,FR,DE,ES,IT,JA,KO,ZHTW).3ds","Pokémon Ultra Sun (EN,FR,DE,ES,IT,JA,KO,ZHTW)",3DS,Pokémon Ultra Sun,更新資料 Ver. 1.2 精靈寶可夢 究極之日,An update for Pokémon™ Ultra Sun is now availa...,Nintendo 3DS遊戲軟體『精靈寶可夢 究極之日』的更新資料。\n持有『精靈寶可夢 究...,Pokémon Ultra Sun
2,1108864,A2AE,3DS,NTSC-U,EN,Pokémon Ultra Sun,Pokémon Ultra Sun,NaN,Pokémon Ultra Sun,Pokémon Ultra Sun,...,1.0,"adventure,role-playing,action rpg",Pokémon Ultra Sun (USA) (EN).3ds,Pokémon Ultra Sun (USA) (EN),3DS,Pokémon Ultra Sun,Pokémon Ultra Sun,NaN,NaN,Pokémon Ultra Sun
3,1108865,A2AJ,3DS,NTSC-J,JP,Pokémon Ultra Sun,Pokémon Ultra Sun,NaN,Pokémon Ultra Sun,Pokémon Ultra Sun,...,1.0,"adventure,role-playing,action rpg",Pokémon Ultra Sun (Japan) (JA).3ds,Pokémon Ultra Sun (Japan) (JA),3DS,Pokémon Ultra Sun,Pokémon Ultra Sun,NaN,NaN,Pokémon Ultra Sun
4,1108866,A2AK,3DS,NTSC-K,KR,Pokémon Ultra Sun,Pokémon Ultra Sun,NaN,Pokémon Ultra Sun,Pokémon Ultra Sun,...,1.0,"role-playing,action rpg",Pokémon Ultra Sun (Korea) (KO).3ds,Pokémon Ultra Sun (Korea) (KO),3DS,Pokémon Ultra Sun,Pokémon Ultra Sun,NaN,NaN,Pokémon Ultra Sun


In [46]:
gamestdb.columns

Index(['N°', 'gametdb_id', 'gametdb_type', 'gametdb_region',
       'gametdb_languages', 'gametdb_title_de', 'gametdb_title_en',
       'gametdb_title_es', 'gametdb_title_fr', 'gametdb_title_it',
       'gametdb_title_ja', 'gametdb_title_ko', 'gametdb_title_nl',
       'gametdb_title_pt', 'gametdb_title_ru', 'gametdb_synopsis_de',
       'gametdb_synopsis_en', 'gametdb_synopsis_es', 'gametdb_synopsis_fr',
       'gametdb_synopsis_it', 'gametdb_synopsis_ja', 'gametdb_synopsis_ko',
       'gametdb_synopsis_nl', 'gametdb_synopsis_pt', 'gametdb_synopsis_ru',
       'gametdb_developer', 'gametdb_publisher', 'gametdb_date',
       'gametdb_players', 'gametdb_genre', 'gametdb_rom',
       'gametdb_rom_sans_ext', 'gametdb_platform', 'gametdb_title_cn',
       'gametdb_title_tw', 'gametdb_synopsis_cn', 'gametdb_synopsis_tw',
       'gametdb_rom_cleaned'],
      dtype='object')

In [47]:
gamestdb.isnull().sum()

N°                          0
gametdb_id                  0
gametdb_type                0
gametdb_region           6400
gametdb_languages        2466
gametdb_title_de          912
gametdb_title_en          918
gametdb_title_es        44811
gametdb_title_fr          848
gametdb_title_it          913
gametdb_title_ja          250
gametdb_title_ko          786
gametdb_title_nl          917
gametdb_title_pt          918
gametdb_title_ru          918
gametdb_synopsis_de     34824
gametdb_synopsis_en     34997
gametdb_synopsis_es     45089
gametdb_synopsis_fr     34696
gametdb_synopsis_it     34935
gametdb_synopsis_ja     33357
gametdb_synopsis_ko     34795
gametdb_synopsis_nl     34922
gametdb_synopsis_pt     34988
gametdb_synopsis_ru     34994
gametdb_developer       30586
gametdb_publisher       16140
gametdb_date                0
gametdb_players          8457
gametdb_genre           21728
gametdb_rom               918
gametdb_rom_sans_ext        6
gametdb_platform            0
gametdb_ti

In [48]:
# show distinct gametdb_type values
gamestdb['gametdb_type'].unique()

array(['3DS', 'DS', 'PS3', 'Switch', 'Wii', 'WiiU'], dtype=object)

Gametdb analysis: nice dataset but only 6 platforms represented. We're still going to use it.

## Launchbox

In [49]:
launchbox = pd.read_csv('csv/launchbox.csv', low_memory=False)
print(launchbox.shape[0])
launchbox.head()

164347


,N°,DatabaseID,Name,ReleaseYear,Overview,MaxPlayers,ReleaseType,Cooperative,VideoURL,CommunityRating,...,AlternateName_China,AlternateName_Europe,AlternateName_France,AlternateName_Germany,AlternateName_Japan,AlternateName_Korea,AlternateName_NorthAmerica,AlternateName_Spain,AlternateName_UnitedStates,AlternateName_World
0,6658890,17687,20th Century Video Almanac,1993.0,"In The Best of Our Century, we've taken multim...",1.0,Released,0,https://www.youtube.com/watch?v=dfFT6jqjFq0,3.301418,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,6658891,219683,3D Atlas,1994.0,The World Isn't Flat. Why Should Your Atlas Be...,1.0,Released,0,https://www.youtube.com/watch?v=RZ-pph55Ui4,3.500000,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,6658892,25182,3DO Action Pak,1995.0,This is a four-game compilation pack that cont...,1.0,Released,0,NaN,3.647436,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,6658893,417702,3DO de Shiru Miru Asobu Nakajima Miyuki,NaN,NaN,1.0,Unreleased,0,NaN,3.750000,...,NaN,NaN,NaN,NaN,3DOで知る・みる・あそぶ 中島みゆき,NaN,NaN,NaN,NaN,3DO de Shiru Miru Asobu - Nakajima Miyuki
4,6658894,130908,3DO Demo Disc Program,NaN,A white binder with blue silk-screened art. Th...,NaN,NaN,0,NaN,3.330357,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [50]:
launchbox['Platform'].unique()

array(['3DO Interactive Multiplayer', 'Amstrad CPC', 'Android',
       'Aamber Pegasus', 'Acorn Archimedes', 'Acorn Atom',
       'Acorn Electron', 'Amstrad GX4000', 'Apple II', 'Arcade',
       'APF Imagination Machine', 'Apogee BK-01', 'Apple IIGS',
       'Apple iOS', 'Apple Mac OS', 'Atari 2600', 'Atari 5200',
       'Atari 800', 'Atari 7800', 'Atari Jaguar', 'Atari ST',
       'Atari Jaguar CD', 'Atari Lynx', 'BBC Microcomputer System',
       'ColecoVision', 'Commodore 128', 'Commodore 64', 'Atari XEGS',
       'Bally Astrocade', 'Bandai Super Vision 8000', 'Camputers Lynx',
       'Casio Loopy', 'Casio PV-1000', 'Coleco ADAM', 'Commodore Amiga',
       'Commodore Amiga CD32', 'Commodore CDTV', 'Commodore MAX Machine',
       'Commodore PET', 'Commodore VIC-20', 'Epoch Super Cassette Vision',
       'Exelvision EXL 100', 'Fujitsu FM Towns Marty', 'Microsoft MSX',
       'Commodore Plus 4', 'Dragon 32/64', 'EACA EG2000 Colour Genie',
       'Elektronika BK', 'Emerson Arcadia 2001'

In [51]:
launchbox.isnull().sum()

N°                                 0
DatabaseID                         0
Name                               0
ReleaseYear                   102778
Overview                       12720
MaxPlayers                     42462
ReleaseType                    10636
Cooperative                        0
VideoURL                       91630
CommunityRating                53271
Platform                           0
ESRB                           45596
CommunityRatingCount               0
Genres                         19421
Developer                      16432
Publisher                      15768
WikipediaURL                  121580
AlternateName_Australia       164198
AlternateName_Brazil          164125
AlternateName_China           162602
AlternateName_Europe          160568
AlternateName_France          163093
AlternateName_Germany         162180
AlternateName_Japan           154593
AlternateName_Korea           163759
AlternateName_NorthAmerica    162715
AlternateName_Spain           162274
A

Launchbox analysis: huge list, some missing values. Prime candidate for the main dataset that other datasets will be added to.

## Mobygames

In [52]:
mobygames = pd.read_csv('csv/mobygames.csv')
print(mobygames.shape[0])
mobygames.head()

316257


,numauto,platform_id,platform_name,game_id,game_title,release_date,mobygames_url
0,2781406,35,3do,96,Quarantine,NaN,t1H5j95PeMaZo+Cw1R9vKw==hY9Nu9IGxGC1KpiwIcK0pX...
1,2781407,35,3do,111,Virtuoso,NaN,pskb3XM01ASHZRWZmUS00g==9/zl7gK72ixL8RSmDQYsDq...
2,2781408,35,3do,165,Princess Maker 2,NaN,IaammuI4ElexSjqcTb6zAg==LU77bhsLWm+TMdKjgrt5Wv...
3,2781409,35,3do,179,Star Control II,NaN,5NOpZrNp7xOTX79DuN4weg==uPkHQmlKhUSynxAzrMmt4b...
4,2781410,35,3do,210,Cannon Fodder,NaN,ICShm7ilii6jmpRScaapPw==QXCpV2CiXimgLBnmVrOM6c...


In [53]:
mobygames.isnull().sum()

numauto               0
platform_id           0
platform_name         0
game_id               0
game_title            0
release_date     316257
mobygames_url         0
dtype: int64

Mobygames analysis: Only platform names and game titles here and release dates are empty. Huge list 316257 entries. A good candidate to attach other info to.

## Recalbox

In [54]:
recalbox = pd.read_csv('csv/recalbox_gamelist.csv')
print(recalbox.shape[0])
print(recalbox.columns)
recalbox.head()

535
Index(['numauto_rom', 'nomjeu_rom', 'fichier_rom', 'description_rom',
       'image_rom', 'rating_rom', 'annee_rom', 'developer_rom',
       'publisher_rom', 'genre_rom', 'players_rom', 'cache_rom', 'favoris_rom',
       'boxart_rom', 'screenshot_rom', 'wheel_rom', 'video_rom', 'mix_rom',
       'playcount_rom', 'lastplayed_rom', 'md5_rom', 'core_rom',
       'emulator_rom', 'cartridge_rom', 'hash_rom', 'manual_rom', 'region_rom',
       'thumbnail_rom', 'marquee_rom', 'genreid_rom', 'fanart_rom', 'map_rom',
       'titleshot_rom', 'lang_rom', 'gameid_rom', 'kid_rom', 'adult_rom',
       'arcadesystemname_rom', 'gametime_rom', 'boxback_rom', 'temporary_rom',
       'bezel_rom', 'ratio_rom', 'rotation_rom', 'extra1_rom', 'famille_rom',
       'mode_rom'],
      dtype='object')


,numauto_rom,nomjeu_rom,fichier_rom,description_rom,image_rom,rating_rom,annee_rom,developer_rom,publisher_rom,genre_rom,...,arcadesystemname_rom,gametime_rom,boxback_rom,temporary_rom,bezel_rom,ratio_rom,rotation_rom,extra1_rom,famille_rom,mode_rom
0,33308,4-in-1 Fun Pak Volume II,"./4-in-1 Fun Pak Volume II (USA, Europe).zip",Now you can have four of the most popular boar...,"./images/4-in-1 Fun Pak Volume II (USA, Europe...",0.4,19920901T000000,Beam Software,Interplay,Variados-Jogo de tabuleiro,...,NaN,36,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,33310,The Addams Family,"./Addams Family, The (USA).zip",The Addams Family is a side-scrolling platform...,"./images/Addams Family, The (USA).png",0.4,19920102T000000,Ocean,Ocean,Ação,...,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,33312,Adventure Island II: Aliens in Paradise,./Adventure Island II - Aliens in Paradise (US...,Jeannie Jungle has been captured by aliens! Sh...,./images/Adventure Island II - Aliens in Parad...,0.4,19930202T000000,Hudson,Hudson,Ação,...,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,33314,The Adventures of Star Saver,"./Adventures of Star Saver, The (USA, Europe).zip","On a routine check through the city, police of...","./images/Adventures of Star Saver, The (USA, E...",0.4,19920101T000000,A-Wave,Taito,Ação-Plataforma,...,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,33315,Aerostar,"./Aerostar (USA, Europe).zip","Aerostar is a top-down, vertically-scrolling s...","./images/Aerostar (USA, Europe).png",1.0,19910901T000000,Sigma Pro-Tech,"Vic Tokai, Inc.",Shoot'em Up-Ação,...,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [55]:
# players_rom values
recalbox['players_rom'].value_counts()

players_rom
1      358
1-2    148
1-4     15
2       12
Name: count, dtype: int64

In [56]:
# rating max min
print(recalbox['rating_rom'].max())
print(recalbox['rating_rom'].min())

1.0
0.0


In [57]:
recalbox['genre_rom'].unique()

array(['Variados-Jogo de tabuleiro', 'Ação', 'Ação-Plataforma',
       "Shoot'em Up-Ação", 'Quebra-Cabeças', 'Esporte', 'Quebra-cabeças',
       "Shoot'em Up", 'Ação-Quebra-Cabeças', 'Aventura', 'Plataforma',
       'Luta-Ação', 'Ação-Estratégia', 'Briga De Rua-Ação',
       'Plataforma-Ação', 'Quebra-Cabeças-Ação',
       'Quebra-Cabeças-Jogo De Tabuleiro', 'Variados',
       'Quebra-cabeças-Estratégia', "Quebra-Cabeças-Ação-Shoot'em Up",
       'Estratégia', 'Simulação-Aventura', 'Quebra-cabeças-Ação',
       'Ação-Aventura', 'Corrida, Pilotagem', 'Action, Platform',
       'Briga De Rua', 'Plataforma / Combate Com Rolagem-Plataforma',
       'Plataforma-Corrida, Pilotagem-Ação', 'Puzzle',
       'Esporte / Futebol-Esporte', nan, 'Simulação', 'Racing, Driving',
       'Esporte-Simulação',
       'Fighting / 2D, Fighting, Platform / Fighter Scrolling, Platform',
       'Jogos De Rpg', 'Jogos de RPG', 'Esporte-Ação', 'Tiro-Ação',
       'Plataforma / Combate Com Rolagem-Ação-Plataforma

Recalbox: Looks like its (mostly) NES/Famicom titles. It has descriptions. Looks like the xml version has less missing values. Let's try joining these two before merging them to main.

## Metacritic

In [58]:
all_games = pd.read_csv('csv/all_games.csv')
print(all_games.shape[0])
all_games.head()

18800


,name,platform,release_date,summary,meta_score,user_review
0,The Legend of Zelda: Ocarina of Time,Nintendo 64,"November 23, 1998","As a young boy, Link is tricked by Ganondorf, ...",99,9.1
1,Tony Hawk's Pro Skater 2,PlayStation,"September 20, 2000",As most major publishers' development efforts ...,98,7.4
2,Grand Theft Auto IV,PlayStation 3,"April 29, 2008",[Metacritic's 2008 PS3 Game of the Year; Also ...,98,7.7
3,SoulCalibur,Dreamcast,"September 8, 1999","This is a tale of souls and swords, transcendi...",98,8.4
4,Grand Theft Auto IV,Xbox 360,"April 29, 2008",[Metacritic's 2008 Xbox 360 Game of the Year; ...,98,7.9


In [59]:
print(all_games['platform'].unique())

[' Nintendo 64' ' PlayStation' ' PlayStation 3' ' Dreamcast' ' Xbox 360'
 ' Wii' ' Xbox One' ' PC' ' Switch' ' PlayStation 2' ' PlayStation 4'
 ' GameCube' ' Xbox' ' Wii U' ' Game Boy Advance' ' 3DS' ' Xbox Series X'
 ' DS' ' PlayStation Vita' ' PlayStation 5' ' PSP' ' Stadia']


In [60]:
all_games.isnull().sum()

name              0
platform          0
release_date      0
summary         114
meta_score        0
user_review       0
dtype: int64

all_games: really nice list. barely any missing values. definitely useful.

### game_dataset_cleaned.csv

via google dataset search. i think its igdb

In [61]:
game_dataset = pd.read_csv('csv/game_dataset_cleaned.csv')
print(game_dataset.shape[0])
print(game_dataset.info())

232595
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 232595 entries, 0 to 232594
Data columns (total 23 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   id                     232595 non-null  int64  
 1   name                   232595 non-null  object 
 2   category               232595 non-null  object 
 3   release_date           179516 non-null  object 
 4   rating                 28787 non-null   float64
 5   aggregated_rating      13709 non-null   float64
 6   genres                 211627 non-null  object 
 7   themes                 143305 non-null  object 
 8   franchise              15118 non-null   object 
 9   series                 31377 non-null   object 
 10  main_developers        102113 non-null  object 
 11  supporting_developers  2492 non-null    object 
 12  publishers             99841 non-null   object 
 13  platforms              193690 non-null  object 
 14  player_perspectives    93537 

In [62]:
game_dataset.head()

,id,name,category,release_date,rating,aggregated_rating,genres,themes,franchise,series,...,platforms,player_perspectives,game_modes,game_engines,similar_games,keywords,cover_url,summary,screenshot_urls,artwork_urls
0,1,Thief II: The Metal Age,main_game,2000-03-21,87.197770,90.000000,"['Shooter', 'Simulator', 'Adventure']","['Action', 'Fantasy', 'Stealth']",NaN,['Thief'],...,['PC (Microsoft Windows)'],['First person'],['Single player'],['Dark Engine'],"[2, 3, 47823, 4, 55038, 9377, 105049, 36198, 4...","['ghosts', 'thief', 'death', 'swordplay', 'ste...",https://images.igdb.com/igdb/image/upload/t_co...,The ultimate thief is back! Tread softly as yo...,['https://images.igdb.com/igdb/image/upload/t_...,['https://images.igdb.com/igdb/image/upload/t_...
1,2,Thief: The Dark Project,main_game,1998-11-30,86.633969,70.000000,"['Simulator', 'Adventure']","['Action', 'Fantasy', 'Stealth']",NaN,['Thief'],...,['PC (Microsoft Windows)'],['First person'],['Single player'],['Dark Engine'],"[1, 533, 3, 4, 11118, 3025, 5647, 41, 9377, 3188]","['ghosts', 'assassin', 'basketball', 'medieval...",https://images.igdb.com/igdb/image/upload/t_co...,Thief is a first-person stealth game that like...,['https://images.igdb.com/igdb/image/upload/t_...,['https://images.igdb.com/igdb/image/upload/t_...
2,3,Thief: Deadly Shadows,main_game,2004-05-25,81.896309,83.000000,"['Shooter', 'Simulator', 'Adventure']","['Action', 'Fantasy', 'Stealth']",NaN,['Thief'],...,"['Xbox', 'PC (Microsoft Windows)']","['First person', 'Third person']",['Single player'],['Unreal Engine 1'],"[1, 9377, 2, 11118, 19441, 4, 533, 3025, 41, 9...","['ghosts', 'assassin', 'minigames', 'thief', '...",https://images.igdb.com/igdb/image/upload/t_co...,"In the third instalment of the Thief series, m...",['https://images.igdb.com/igdb/image/upload/t_...,NaN
3,4,Thief,main_game,2014-02-25,69.898552,63.642857,"['Shooter', 'Adventure']","['Action', 'Stealth', 'Sandbox']",NaN,['Thief'],...,"['PlayStation 3', 'PlayStation 4', 'PC (Micros...",['First person'],['Single player'],['Unreal Engine 3'],"[5647, 3188, 1121, 3025, 1877, 19441, 9377, 1,...","['thief', 'steampunk', 'blackjack', 'action-ad...",https://images.igdb.com/igdb/image/upload/t_co...,There is a rising tide of fear in The City. Ha...,['https://images.igdb.com/igdb/image/upload/t_...,['https://images.igdb.com/igdb/image/upload/t_...
4,5,Baldur's Gate,main_game,1998-12-21,85.799533,NaN,['Role-playing (RPG)'],"['Fantasy', 'Open world']",['Dungeons & Dragons'],"[""Baldur's Gate""]",...,"['Linux', 'PC (Microsoft Windows)', 'iOS', 'Mac']",['Bird view / Isometric'],"['Single player', 'Co-operative']",['Infinity Engine'],"[19164, 19698, 81249, 103303, 23651, 49414, 11...",NaN,https://images.igdb.com/igdb/image/upload/t_co...,Baldur's Gate is a fantasy role-playing video ...,['https://images.igdb.com/igdb/image/upload/t_...,NaN


In [67]:
# explode platforms values and count distinct platforms
game_dataset['platforms'] \
    .dropna() \
    .str.strip('[]') \
    .str.replace("'", "", regex=False) \
    .str.split(',') \
    .explode() \
    .str.strip() \
    .value_counts()

platforms
PC (Microsoft Windows)    128267
Mac                        25340
Linux                      16588
iOS                        15774
Nintendo Switch            15455
                           ...  
AY-3-8605                      1
Amstrad GX4000                 1
Exidy Sorcerer                 1
Virtual Console                1
Polymega                       1
Name: count, Length: 215, dtype: int64

Looks good. Developer, platform and publisher are stored as lists so thats something we have to deal with downstream. 

We will use this dataset. 

### Exploration Results

Launchbox will be used as primary

To be merged:

- Metacritic (all_games)
- GamesTDB
- DAT_DOS
- DAT_MAME
- Recalbox
- Game_dataset_cleaned



